nav2_dwb_controller/dwb_core/src/publisher.cpp

```diff
@@ -146,18 +146,20 @@ DWBPublisher::on_cleanup()
 void
 DWBPublisher::publishEvaluation(std::shared_ptr<dwb_msgs::msg::LocalPlanEvaluation> results)
 {
+  if (publish_evaluation_ && node_->count_subscribers(eval_pub_->get_topic_name()) < 1) {return;}
```

Comment:
This is incorrect - its not and its or, but these are also different things that don't necessarily need to be compounded. Leave the `publish_evaluation_` check wrapping the actual `->publish(..)` itself.

Reply:
Didn't saw `publish_evaluation` there!